DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/hardware_reuse/loop/src/main.cpp

```diff
@@ -0,0 +1,132 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class VectorOpID;
+
+using D3Vector = std::array<float, 3>;
+
+// Minimum capacity of a pipe.
+// Set to 0 to allow the compiler to save area if possible.
+constexpr size_t kPipeMinCapacity = 0;
+
+// Pipes
+class IDInputPipeA;
+class IDOutputPipeZ;
+
+using InputPipeA = sycl::ext::intel::experimental::pipe<IDInputPipeA, D3Vector,
+                                                        kPipeMinCapacity>;
+using OutputPipeZ =
+    sycl::ext::intel::experimental::pipe<IDOutputPipeZ, float, 5>;
```

Comment:
```suggestion
    sycl::ext::intel::experimental::pipe<IDOutputPipeZ, float, kPipeMinCapacity>;
```

Reply:
We probably can take advantage of capacity balancing here and another 2 versions.